src/main/java/com/ibm/nfvodriver/service/LifecycleManagementService.java

```diff
@@ -40,72 +42,60 @@ public LifecycleManagementService(NSLifecycleManagementDriver nsLifecycleManagem
 
     public ExecutionAcceptedResponse executeLifecycle(ExecutionRequest executionRequest) throws MessageConversionException {
         logger.info("Processing execution request");
-
+        String lifecycleName = executionRequest.getLifecycleName();
+        final String requestId = UUID.randomUUID().toString();
         try {
-            if ("Create".equalsIgnoreCase(executionRequest.getLifecycleName())) {
-                final String requestId = UUID.randomUUID().toString();
-                // Generate CreateNSRequest message
-                final String createNsRequest = messageConversionService.generateMessageFromRequest("CreateNsRequest", executionRequest);
```

Comment:
generateMessageFromRequest method call is there in all cases, can't we keep ahead of having switch and pass the 1st parameter based on the lifecycle, fetch the real value from a map.